lib/puppet/provider/docker_compose/ruby.rb

```diff
@@ -0,0 +1,29 @@
+Puppet::Type.type(:docker_compose).provide(:ruby) do
+  desc "Support for Docker Compose"
+  commands :dockercompose => "docker-compose"
+    
+  mk_resource_methods
+
+
+  def exists?
+    Puppet.info("Checking if docker-compose.yml exists")
+    begin
+      Dir.chdir(resource[:source])
+      File.file?('docker-compose.*')    
+    rescue Exception => e
```

Comment:
the exception is stored in `e` here but is then not used. It's probably best to do this without an exception as well, as not finding a file isn't _exceptional_.